Assets/LoggingManager/LogStore.cs

```diff
@@ -0,0 +1,276 @@
+﻿using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Boo.Lang.Runtime;
+using UnityEngine;
+
+
+public enum LogType
+{
+    Normal,
+    Meta
+}
+
+public class LogStore : MonoBehaviour
+{
+
+    private SortedDictionary<string, List<string>> logs;
+    private StringBuilder logString;
+    public string Label { get; set; }
+
+    private LogType logType;
+    private bool logCommonColumns;
+
+    private int nbLines;
+    private bool createStringOverTime;
+    private StringBuilder currentLineLogged;
+    private SortedDictionary<string, string> currentLogs;
+    private const string fieldSeparator = ";";
+    private const string lineSeperator = "\n";
+    private string email;
+    public string SessionId { get; set; }
+
+    public LogStore(string label, string email, bool createStringOverTime,
+        LogType logType = LogType.Normal, bool logCommonColumns = true)
+    {
+        Init(label, email, Guid.NewGuid().ToString(), createStringOverTime, logType, logCommonColumns);
+    }
```

Comment:
Could we set the `createStringOverTime` bool to `true` by default? Would that work for e.g. Whack-A-Mole etc.? Either way it would be good to settle on a default value, so people dont have to think of it.

What is the `Guid.NewGuid()` function? Is it some built-in way to generate IDs? What is it using internally?

Reply:
### "Could we set the createStringOverTime bool to true by default?"
The `createStringOverTime` bool is set to true in the LoggingManager [here](https://github.com/Vincent-SD/ArduinoLogger/blob/06cbd9a1d1288205624b9d286d0c12b86799b5c6/Assets/LoggingManager/LoggingManager.cs#L35).

The user don't have to pay attention to this value because it is already defined in the manager and the LogStore objects are only created by this manager. I decided not to put a default value in the LogStore class to make the code more adaptable if one day it is decided to put some parts of this code in another project.

### "Would that work for e.g. Whack-A-Mole etc.?"
It will not work everywhere.  The log structure must be clean and the program cannot add headers during the logging process, otherwise this feature will be disabled. 

### "What is the Guid.NewGuid() function? Is it some built-in way to generate IDs? What is it using internally?"

This function is used to generate a random unique ID, so that each log will have its own id even if the user doesn't give it one.

Reply:
Fine with me, thanks for the explanation.
Also, fine with using `Guid.NewGuid()`, I think it's better than us making up our own way of making IDs.